start TS readme (#380)
GetStream/stream-chat-react-native: README.md

- [Internationalisation (i18n)](#internationalisation)
- [Cookbook](https://github.com/GetStream/stream-chat-react-native/wiki/Cookbook)  :rocket:

## Contents

- [React Native Compatibility](#react-native-compatibility)
- [React Native Chat Tutorial](#react-native-chat-tutorial)
- [Example Apps](#example-apps)
  - [Expo example](#expo-example)
  - [Slack clone](#slack-clone)
- [Docs](#docs)
- [Keep in mind](#keep-in-mind)
- [Setup](#setup-setting-up-a-chat-app)
  - [Expo package](#expo-package)
  - [Native package](#native-package)
  - Native Web package _(currently under development. Please follow progress at [#206](https://github.com/GetStream/stream-chat-react-native/issues/206))_
  - [TypeScript Support](#typescript-support)
  - [Upgrade](#upgrade)
  - [Common issues](#common-issues)
  - [Contributing](#contributing)

## React Native Compatibility

To use this library you need to ensure you match up with the correct version of React Native you are using.

| `stream-chat-react-native` version | Required React Native Version |
| ----------------------------------------- | --------- |

> | `2.x.x`                                   | `>= 0.59` |

Author: agreed

### Native example

1. Please make sure you have installed necessary dependencies depending on your development OS and target OS. Follow the guidelines given on official React Native documentation for installing dependencies: <https://facebook.github.io/react-native/docs/getting-started>#
2. Make sure node version is >= v10.13.0
3. Start the simulator

4. ```bash
   git clone https://github.com/GetStream/stream-chat-react-native.git
   cd stream-chat-react-native

>    yarn install
>    cd stream-chat-react-native/native-package
>    yarn install
>    cd stream-chat-react-native/examples/NativeMessaging
>    yarn install

Reviewer: ```suggestion
   yarn
   cd stream-chat-react-native/native-package
   yarn
   cd stream-chat-react-native/examples/NativeMessaging
   yarn
```

   ```

5. - For iOS

     ```bash
     cd ios && pod install && cd ..

>      npx react-native run-ios

Reviewer: ```suggestion
     yarn ios
```

     ```

   - For android

     ```bash

>      npx react-native run-android

Reviewer: ```suggestion
     yarn android
```

## Common issues

#### While running native example, you may (not necessarily) run into the following issues

> 1. When you execute `npx react-native run-ios` for the first time, it starts a metro bundler in parallel. It can result in some errors since the build process isn't complete yet. Try the following to fix this:

Reviewer: ```suggestion
1. When you execute `yarn ios` for the first time, it starts a metro bundler in parallel. It can result in some errors since the build process isn't complete yet. Try the following to fix this:
```

   1. Close/stop the metro bundler process.
   2. Let the build process finish. It can usually take around 2-3 minutes for the first time.
   3. Start the metro bundler manually by executing `yarn start` inside `stream-chat-react-native/examples/NativeMessaging` directory.

> 2. When you execute `npx react-native run-android`, you may (not necessarily) run into following error:

Reviewer: ```suggestion
2. When you execute `yarn android`, you may (not necessarily) run into following error:
```

   1. Create a file named `local.properties` inside `stream-chat-react-native/examples/NativeMessaging/android` directory
   2. Put the this line in that file. Make sure sdk path is correctly mentioned as per your system:

      ```
      sdk.dir=/Users/{user_name}/Library/Android/sdk/
      ```

>    3. Rerun `npx react-native run-android` in `stream-chat-react-native/examples/NativeMessaging` directory

Reviewer: ```suggestion
   3. Rerun `yarn android` in `stream-chat-react-native/examples/NativeMessaging` directory
```
